### ISSUE TITLE
Check if a user is LIVE or OFFLINE

### ISSUE THREAD

**Issue by bofinbabu** [justintv/Twitch-API#548]
I'm entirely new to twitch API.
From what I've read from the documentation so far, I'm unclear how I'm going to make this simple API call which will tell if a user is streaming live or not?
Note: In my particular case, the user is myself.

**Comment by Decicus** [comment]
If you make a call to [/streams/:channel/](https://github.com/justintv/Twitch-API/blob/master/v3_resources/streams.md#get-streamschannel), `stream` will either be an object (if LIVE) or `null` (if OFFLINE).

**Comment by bofinbabu** [comment]
Yes it's working. But it's taking about 20 seconds recognize the change (took 20 sec for that api call to return a null, instead of the object, after I stopped broadcasting)
Is it normal?

**Comment by bashtech** [comment]
That is normal.

**Comment by bofinbabu** [comment]
@bashtech @Decicus 
It's not just 20 seconds.
It's about **1 minute 45 seconds** to detect that the stream has started.
And it's about **3 mins** to detect that the broadcasting stopped.

Is there any way to reduce these delays?

**Comment by TournyMasterBotCurse** [comment]
The time will vary, the endpoints are cached to improve performance, this is expected behavior.

**Comment by jasongornall** [comment]
bumping this..  is there anyway to speed this up?

**Comment by Marioario** [comment]
@TournyMasterBotCurse how long is the cache?

**Comment by AVANAAR6** [comment]
Long is length, we use words,, C

-----Original Message-----

**Comment by Fugiman** [comment]
No, it can take anywhere up to 3 minutes for streams to change state due to various layers of caching. There is no way for you to speed this up.

**Comment by Argus9** [comment]
@Fugiman what about using event listeners to catch when a channel goes online/offline?

**Comment by mavrick** [comment]
@Argus9 @Fugiman here is a link if you are wondering what the event listeners are using the embeded video player https://github.com/justintv/Twitch-API/blob/master/embed-video.md#events

**Comment by maxtruxa** [comment]
Or you could just use the ([now documented](https://github.com/justintv/Twitch-API/blob/master/PubSub/README.md)) PubSub system.

Listen for `video-playback.some_channel` and look for `stream-up`/`stream-down`. Documentation for the `video-playback` topic is available in [this PR](https://github.com/justintv/Twitch-API/pull/583).
